### PR TITLE
Implement apply_unordered method

### PR DESCRIPTION
**Pull request by jsgoyette**
This PR implements the `apply_unordered` method on the `Validator`.

### CODE REVIEW

File: filmreel/src/response.rs [hunk 1]
```diff
@@ -258,6 +256,73 @@ impl Validator {
...
+                    _ => return Ok(()),
+                };
+
+                // Create a lookup hashmap for self_selection.
+                let mut self_selection_hash: HashMap<Key, bool> = HashMap::new();
+                self_selection.iter().for_each(|s| {
+                    if let Ok(k) = to_key(&s) {
+                        self_selection_hash.insert(k, true);
+                    };
+                });
```
Comment (mkatychev) [inline filmreel/src/response.rs:283]: This pattern will swallow any errors returned by `to_key`.

We can instead call  `.collect()` on a map of results, unwrap it with `?` and call `.collect()` collect again on valid `Key` values _if_ we have no errors:

```rust
// Create a lookup hashmap for self_selection.
let self_selection_hash: HashMap<Key, bool> = self_selection
	.iter()
	.map(to_key) // Result<Key, Error>
	.collect::<Result<Vec<Key>, _>>()
	.map_err(|e| FrError::ReadInstructionf("apply_unordered", e.to_string()))?
	.into_iter() // IntoIterator<Item = Key>
	.map(|k| (k, true))
	.collect();
```

Comment (phernandez) [reply filmreel/src/response.rs:283]: Whoa. I'm having Java8 flashbacks. 😰

Comment (mkatychev) [reply filmreel/src/response.rs:283]: @phernandez  🤷  There's also the go style of having the error be a separate result [using `partition()`](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#collect-all-valid-values-and-failures-with-partition)

Comment (mkatychev) [reply filmreel/src/response.rs:283]:
```rust
let (keys, errors): (Vec<_>, Vec<_>) = self_selection()
        .iter()
         ... // do the map
        .partition(Result::is_ok); // split based on if value is Ok(value)
errors.into_iter().collect()?; // unwrap errors here
```

Comment (mkatychev) [reply filmreel/src/response.rs:283]: Probably a cleaner more readable way of doing it as well 🤔

**Review by mkatychev — APPROVED** [review]
LGTM

**Comment by jsgoyette** [comment]
Merging this now. Mikhail has some improvements to add to this, but they can be added afterwards.

**Pull request merged**